### PR TITLE
Add client and model readmes and fix partition context

### LINKED ISSUE #18907: Partition context binding in new webjob event hub extensions
**Query/Question**
I am not really sure if the new web job event hub extension package is usable yet, but can I still access the partition context when processing a batch of event hub messages? I tried running this piece of code for example

```
public Task MyEventHubListener(
            [EventHubTrigger(
                eventHubName: "",
                Connection = "MyConnection",
                ConsumerGroup = "$Default")] EventData[] eventHubMessages,
            PartitionContext partitionContext,
            CancellationToken cancellationToken)
        {
        }
```

And got this exception:

```
fail: Function.MyEventHubListener[3]
      Executed 'Functions.MyEventHubListener' (Failed, Id=85c28c5b-1fee-4a4d-9846-08151fac91c4, Duration=621ms)
      Microsoft.Azure.WebJobs.Host.FunctionInvocationException: Exception while executing function: Functions.MyEventHubListener
       ---> System.InvalidOperationException: Exception binding parameter 'partitionContext'
       ---> System.InvalidOperationException: Binding data for 'partitionContext' is not of expected type PartitionContext.
         at Microsoft.Azure.WebJobs.Host.Bindings.Data.ClassDataBinding`1.BindAsync(BindingContext context) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Bindings\Data\ClassDataBinding.cs:line 65
         at Microsoft.Azure.WebJobs.Host.Triggers.TriggeredFunctionBinding`1.BindCoreAsync(ValueBindingContext context, Object value, IDictionary`2 parameters) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Triggers\TriggeredFunctionBinding.cs:line 97
         --- End of inner exception stack trace ---
         at Microsoft.Azure.WebJobs.Host.Executors.DelayedException.Throw() in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\DelayedException.cs:line 27
         at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ParameterHelper.PrepareParametersAsync() in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 892
         at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithWatchersAsync(IFunctionInstanceEx instance, ParameterHelper parameterHelper, ILogger logger, CancellationTokenSource functionCancellationTokenSource) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 470
         at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance, FunctionStartedMessage message, FunctionInstanceLogEntry instanceLogEntry, ParameterHelper parameterHelper, ILogger logger, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 283
         --- End of inner exception stack trace ---
         at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance, FunctionStartedMessage message, FunctionInstanceLogEntry instanceLogEntry, ParameterHelper parameterHelper, ILogger logger, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 330
         at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.TryExecuteAsync(IFunctionInstance functionInstance, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 94
```

**Environment:**
 - Name and version of the Library package used: Microsoft.Azure.WebJobs.Extensions.EventHubs 5.0.0-beta.1

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/18907

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/Samples/TriggerSingleModel.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            public int Age { get; set; }
+        }
+
+        #region Snippet:TriggerSingleModel
+        [FunctionName("TriggerSingleModel")]
+        public static void Run(
+            [EventHubTrigger("<event_hub_name>", Connection = "<connection_name>")] Dog dog,
+            ILogger logger)
+        {
+            logger.LogInformation($"Who's a good dog? {dog.Name} is!");
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/Samples/TriggerSingleModel.cs:23]: Nice

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHostPartition.cs [hunk 2]
```diff
@@ -20,17 +22,21 @@ public EventProcessorHostPartition(string partitionId)
             PartitionId = partitionId;
         }
 
+        public PartitionContext PartitionContext => _partitionContext ??= new EventProcessorHostPartitionContext(this);
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHostPartition.cs:25]: Where is _partitionContext set?

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs [hunk 3]
```diff
@@ -385,13 +398,34 @@ public static void SendEvent_TestHub(string input, [EventHub(TestHubName, Connec
...
+        {
+            public static async Task SendEvents([EventHub(TestHubName, Connection = TestHubName)] EventHubProducerClient producer)
+            {
+                await producer.SendAsync(new[]
+                {
+                    new EventData(new BinaryData("Event 1")),
+                });
+            }
+
+            public static void ProcessSingleEvent([EventHubTrigger(TestHubName, Connection = TestHubName)] EventData eventData)
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs:426]: Shouldn't there be an end to end test that binds to PartitionContext?

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs (hunk 3 above)
Comment (pakrym) [reply sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs:426]: I added one in this PR https://github.com/Azure/azure-sdk-for-net/pull/19114/files#diff-ebf190a222702ad012429237484e3b2cc982fec69efad485d34808b90ecd6121R402

**Pull request merged**